chamge/amqp/amqp-rpc.c

```diff
@@ -0,0 +1,700 @@
+      break;
+    case PROP_PORT:
+      g_value_set_uint (value, priv->connection_info.port);
+      break;
+    case PROP_CHANNEL:
+      g_value_set_uint (value, priv->channel);
+      break;
+    case PROP_VHOST:
+      g_value_set_string (value, priv->connection_info.vhost);
+      break;
+    case PROP_ID:
+      g_value_set_string (value, priv->connection_info.user);
+      break;
+    case PROP_PWD:
+      g_value_set_string (value, priv->connection_info.password);
+      break;
+    case PROP_URL:
+      g_value_set_string (value, priv->amqp_url);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+_amqp_parse_url (const gchar * url, rpc_connection_info * const conn_info)
+{
+  g_autofree gchar *s = g_strdup (url);
+  amqp_parse_url (url, conn_info);
+  g_debug ("url parsed : host[%s] vhost[%s]", conn_info->host,
+      conn_info->vhost);
+  g_debug ("    user[%s] pwd[%s] port[%d] ssl[%d]", (conn_info)->user,
+      (conn_info)->password, (conn_info)->port);
+}
+
+static void
+chamge_amqp_rpc_set_property (GObject * object,
+    guint prop_id, const GValue * value, GParamSpec * pspec)
+{
+  ChamgeAmqpRpc *self = CHAMGE_AMQP_RPC (object);
+  ChamgeAmqpRpcPrivate *priv = chamge_amqp_rpc_get_instance_private (self);
+
+  switch ((_ChamgeAmqpRpcProperty) prop_id) {
+    case PROP_HOSTNAME:
+      priv->connection_info.host = g_value_dup_string (value);
+      g_debug ("  set hostname : %s", priv->connection_info.host);
+      break;
+    case PROP_PORT:
+      priv->connection_info.port = g_value_get_uint (value);
+      g_debug ("  set port : %d", priv->connection_info.port);
+      break;
+    case PROP_CHANNEL:
+      priv->channel = g_value_get_uint (value);
+      g_debug ("  set channel : %d", priv->channel);
+      break;
+    case PROP_VHOST:
+      priv->connection_info.vhost = g_value_dup_string (value);
+      g_debug ("  set vhost : %s", priv->connection_info.vhost);
+      break;
+    case PROP_ID:
+      priv->connection_info.user = g_value_dup_string (value);
+      g_debug ("  set id : %s", priv->connection_info.user);
+      break;
+    case PROP_PWD:
+      priv->connection_info.password = g_value_dup_string (value);
+      g_debug ("  set pwd : %s", priv->connection_info.password);
+      break;
+    case PROP_URL:
+      priv->amqp_url = g_value_dup_string (value);
+      _amqp_parse_url (priv->amqp_url, &priv->connection_info);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+chamge_amqp_rpc_class_init (ChamgeAmqpRpcClass * klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->get_property = chamge_amqp_rpc_get_property;
+  object_class->set_property = chamge_amqp_rpc_set_property;
+  object_class->dispose = chamge_amqp_rpc_dispose;
+  object_class->finalize = chamge_amqp_rpc_finalize;
```

Comment:
if there’s no specific reason, let’s override `_dispose()` only. 
Normally, it’s enough to free resources in dispose.